hal/sparcv8leon3/cpu.c

```diff
@@ -58,22 +74,22 @@ int hal_cpuCreateContext(cpu_context_t **nctx, void *start, void *kstack, size_t
 	ctx->o5 = 0xf5555555;
 	ctx->o7 = 0xf7777777;
 
-	ctx->l0 = 0xeeeeeee0;
-	ctx->l1 = 0xeeeeeee1;
-	ctx->l2 = 0xeeeeeee2;
-	ctx->l3 = 0xeeeeeee3;
-	ctx->l4 = 0xeeeeeee4;
-	ctx->l5 = 0xeeeeeee5;
-	ctx->l6 = 0xeeeeeee6;
-	ctx->l7 = 0xeeeeeee7;
-
-	ctx->i0 = (u32)arg;
-	ctx->i1 = 0x10000001;
-	ctx->i2 = 0x10000002;
-	ctx->i3 = 0x10000003;
-	ctx->i4 = 0x10000004;
-	ctx->i5 = 0x10000005;
-	ctx->i7 = (u32)start - 8;
+	wctx->l0 = 0xeeeeeee0;
+	wctx->l1 = 0xeeeeeee1;
+	wctx->l2 = 0xeeeeeee2;
+	wctx->l3 = 0xeeeeeee3;
+	wctx->l4 = 0xeeeeeee4;
+	wctx->l5 = 0xeeeeeee5;
+	wctx->l6 = 0xeeeeeee6;
+	wctx->l7 = 0xeeeeeee7;
+
+	wctx->i0 = 0x10000000;
```

Comment:
How is `arg` now passed to the thread?

Reply:
in `o0` register (`ctx->o0`)

Reply:
Oh, ok, I missed the fact it was assigned two times